src/server/context.ts

```diff
@@ -106,15 +106,28 @@ export class ContextManager implements IContext {
    * @param savedItem
    * @param tagsUpdated tags updated during mutation
    */
-  emitItemEvent(
+  async emitItemEvent(
     event: EventType,
-    savedItem: SavedItem | Promise<SavedItem>,
+    savedItem: SavedItem,
     tagsUpdated?: string[]
-  ): void {
-    this.eventEmitter.emitItemEvent(
-      event,
-      this.generateEventPayload(savedItem, tagsUpdated)
-    );
+  ): Promise<void> {
+    if (savedItem == null) {
+      Sentry.captureEvent({
+        message: 'Save was null or undefined when generating event payload',
+        level: 'warning',
+      });
+      return;
+    }
+    try {
+      const tags = (await this.models.tag.getBySaveId(savedItem.id)).map(
+        (_) => _.name
```

Comment:
Wondering why not use a descriptive variable name instead of `_` 🤔

Reply:
making a note to remove scala idioms in the future